frontend/src/productPages/productPage.component.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import axios from 'axios';
+
+import { apiEndpoints } from '../constants/constants';
+import { Element } from '../interfaces/Element.component';
+import { toastError } from '../notification/ToastNotification.component';
+import styles from '../productPages/productpage.module.css';
+
+export const ProductPage: React.FC = () => {
+  const { id } = useParams<{ id?: string }>();
+  const [product, setProduct] = useState<Element | null>(null);
+
+  useEffect(() => {
+    const fetchProduct = async () => {
+      if (!id) {
+        console.error('Error: Product ID is undefined');
+        return;
+      }
+
+      try {
+        const response = await axios.get(apiEndpoints.products.productID(id));
+        setProduct(response.data);
+      } catch (error) {
+        toastError('Error fetching product:');
+      }
+    };
+
+    fetchProduct();
+  }, [id]);
+
+  const handleButtonClick = () => {
+    if (product?.subtitle) {
+      window.open(product.subtitle, '_blank');
+    } else {
+      toastError('No subtitle link available');
+    }
+  };
+
+  const parsedSpecifications = (() => {
+    if (!product?.specifications || typeof product.specifications !== 'string') {
+      return null;
+    }
+
+    try {
+      return JSON.parse(product.specifications);
+    } catch (error) {
+      toastError('Error parsing specifications');
+      return null;
+    }
+  })();
+
+  if (!product) return <div>Loading...</div>;
```

Comment:
Don't show loading if no product found. Instead, show user some kind of `Product not found` text